helix-core/src/main/java/org/apache/helix/model/ParticipantHistory.java

```diff
@@ -76,7 +80,15 @@ public void reportOffline() {
    * @return
    */
   public void reportOnline(String sessionId, String version) {
-    updateSessionHistory(sessionId, version);
+    String hostname;
+    try {
+      hostname = InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      LOG.error("Failed to get host name. Use {} for the participant history recording.",
```

Comment:
This error indicates the IP of this host cannot be translated to a hostname. This may lead to some connectivity issues. Moreover, the information record in history will be wrong. So I think it is ERROR instead of WARN. If such case happen frequently, then the application admin needs to take a look.